guides/_posts/2014-01-11-ifelse.html

```diff
@@ -36,73 +49,83 @@
 
 {% endhighlight %}
 
-But, any property can be used within an `@if`, `@else` conditional.  So, if you want a layout to be responsive to a panel's width:
+But, any property can be used within an `@if`, `@else` conditional.  So, if you want a layout to be responsive to a section's width:
 
 {% highlight css %}
 
-/* GSS: */
-@if #panel[width] >= 800 {
-  /* ... */
-}
-@else #panel[width] >= 500 {
-  /* ... */
+@if section[width] <= 400 {
+  section {
+    article {
+      :first[top] == ^[top];
+      bottom: == :next[top];
+      left: == ^[left];
+      width: == ^[width];
+      height: == ^[height] / 2;
+    }
+  }
 }
-@else {
-  /* ... */
+@else  {
+  section {
+    article {
+      width: == ^[width] / 2;
+      height: == ^[height];
+      top: == :next[top];
+      right: == :next[left];
+    }
+  }
 }
 
-/* Not possible with Media Queries */
 {% endhighlight %}
 
+When the `section` element `width` goes bellows 400px, the `article` elements will be constrain
+by the `@if` block otherwise by the `@else` block.
+
+### Constraints evaluation in conditions
+In order to determine if some conditions hold, GSS needs to evaluate every constraints outside of those conditions before evaluating if
+they are met, which include constraints appearing after the conditions declaration.
+
+As explained in the [GSS constraints guide](/guide/constraints), the order of constraints declaration can affects the
+resulting solution.
 
+The constraint within the condition of the following example will be added to the solver after the outside constraints. Which
+make sens since the condition is true, you want the constraint within that condition to overcome the other evenly strenghted ones.
 
 {% highlight css %}
 
-@if #panel[width] >= 500 {
-  @horizontal .item;  
-  #button {
-    font-size: == 18;
-  }
-}
-@else {
-  @vertical .item;
-  #button {
-    font-size: == 14;
-  }
+#elm[width] == ::window[width];
+
+@if elm[width] < 400 {
+  #elm[height] == 300;
 }
-{% endhighlight %}
 
+#elm[width] == 300;
+#elm[height] == 500;
```

Comment:
The use of 2 width constraints here is redundant in relation to the point about conditionals. Can we remove one of them to keep things simple?